### PR TITLE
docs(a11y): Adding accessibility sections to datepicker, menu, slide …

### PR DESCRIPTION
**Pull request by tinayuangao**
…toggle, slider, snackbar and toolbar

### CODE REVIEW

File: src/lib/snack-bar/snack-bar.md [hunk 1]
```diff
@@ -72,3 +72,7 @@ export class MessageArchivedComponent {
   constructor(@Inject(MD_SNACK_BAR_DATA) public data: any) { }
 }
 ```
+### Accessibility
+A Snackbar automatically goes away after a short time, so you can't count on the user seeing the
```
Comment (kara) [inline src/lib/snack-bar/snack-bar.md]: Nit: To be consistent with the rest of the doc, use `snack-bar` instead of `Snackbar` ?

Comment (tinayuangao) [reply src/lib/snack-bar/snack-bar.md]: Changed to `snack-bar`. Thanks for review.

**Comment by kara** [comment]
LGTM

File: src/lib/datepicker/datepicker.md [hunk 2]
```diff
@@ -228,3 +228,6 @@ application root module.
 })
 export class MyApp {}
 ```
+### Accessibility
+The input box for datepicker should have a placeholder or be given a meaningful label via
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: I'd like to expand this to describe what the datepicker does in general (input with aria-haspopup that triggers a `role="dialog"`) along with the keyboard shortcuts, but that can go in another PR.

File: src/lib/datepicker/datepicker.md [hunk 3]
```diff
@@ -228,3 +228,6 @@ application root module.
...
+`aria-label` or `aria-labelledby`.
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: We should mention that `MdDatepickerIntl` includes strings that are used for `aria-label`s.

File: src/lib/snack-bar/snack-bar.md [hunk 4]
```diff
@@ -72,3 +72,7 @@ export class MessageArchivedComponent {
   constructor(@Inject(MD_SNACK_BAR_DATA) public data: any) { }
 }
 ```
+### Accessibility
+A snack-bar automatically goes away after a short time, so you can't count on the user seeing the
```
Comment (jelbourn) [inline src/lib/snack-bar/snack-bar.md]: I would phrase this as
```markdown
Snack-bar messages are announced via an `aria-live` region. Focus is not moved to
the snack-bar element, as this would be disruptive to a user in the middle of a
workflow. For any action offered in the snack-bar, the application should offer the
user an alternative way to perform the action (typically via keyboard shortcut).
```

File: src/lib/toolbar/toolbar.md [hunk 5]
```diff
@@ -49,3 +49,7 @@ easily accomplished with `display: flex`:
 The color of a `<md-toolbar>` can be changed by using the `color` property. By default, toolbars
 use a neutral background color based on the current theme (light or dark). This can be changed to 
 `'primary'`, `'accent'`, or `'warn'`.  
+
+### Accessibility
+Toolbar items without text or labels should be given a meaningful label via `aria-label` or
```
Comment (jelbourn) [inline src/lib/toolbar/toolbar.md]: I think this isn't quite right. Similar to `list` and `grid-list`, we treat toolbar as decorative-only by default. Generally, the toolbar is used as a header where `role="header"` would be appropriate. 

Only if the use-case of the toolbar match that of [`role="toolbar"`](https://www.w3.org/TR/wai-aria/roles#toolbar), the user should add the role and an appropriate label.

File: src/lib/datepicker/datepicker.md [hunk 6]
```diff
@@ -228,3 +228,48 @@ application root module.
...
 ```
+### Accessibility
+The `MdDatepickerInput` directive adds `aria-haspopup` attribute to the native input element, and it
+triggers a calendar dialog with `role="dialog"`.
+`MdDatepickerIntl` includes strings that are used for `aria-label`s. The datepicker input
+should have a placeholder or be given a meaningful label via `aria-label` or `aria-labelledby`.
+
+#### Keyboard shortcuts
+The keyboard shortcuts to handle datepicker are:
+`ALT` + `DOWN_ARROR`: trigger calendar dialog
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: `ARROR` -> `ARROW`

**Comment by tinayuangao** [comment]
Comments addressed. Please take another look. Thanks!

File: src/lib/datepicker/datepicker.md [hunk 7]
```diff
@@ -228,3 +228,55 @@ application root module.
...
 ```
+### Accessibility
+The `MdDatepickerInput` directive adds `aria-haspopup` attribute to the native input element, and it
+triggers a calendar dialog with `role="dialog"`.
+
+`MdDatepickerIntl` includes strings that are used for `aria-label`s. The datepicker input
+should have a placeholder or be given a meaningful label via `aria-label`, `aria-labelledby` or
+`MdDatepickerIntl`.
+
+#### Keyboard shortcuts
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: It would be good to put these in a table, e.g.
```markdown
| Shortcut             | Action                    |
|----------------------|---------------------------|
| `ALT` + `DOWN_ARROW` | Open the calendar pop-up  |
| `ESCAPE`             | Close the calendar pop-up |

```

File: src/lib/datepicker/datepicker.md [hunk 8]
```diff
@@ -228,3 +228,55 @@ application root module.
...
+The keyboard shortcuts to handle datepicker are:
+
+`ALT` + `DOWN_ARROR`: trigger calendar dialog
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: `ARROR` -> `ARROW`

File: src/lib/datepicker/datepicker.md [hunk 9]
```diff
@@ -228,3 +228,52 @@ application root module.
...
+
+
+In month view:
+
+| Shortcut             | Action                            |
+|----------------------|-----------------------------------|
+| `LEFT_ARROW`         | Goes to previous day              |
+| `RIGHT_ARROW`        | Goes to next day                  |
+| `UP_ARROW`           | Goes to previous 7 days           |
+| `DOWN_ARROW`         | Goes to next 7 days               |
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: `Goes` -> `Go` for each of these

For up and down arrow, I'd say `Go to same day in the next week`

**Comment by mmalerba** [comment]
LGTM on datepicker and slider

**Comment by tinayuangao** [comment]
Put shortcuts in tables. Please take another look. Thanks!

File: src/lib/datepicker/datepicker.md [hunk 10]
```diff
@@ -228,3 +228,52 @@ application root module.
...
+| `HOME`               | Focus the first day of the month  |
+| `END`                | Focus the last day of the month   |
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker.md]: Why say `Focus` for home/end and `Goes` / `Go` everywhere else?

**Comment by tinayuangao** [comment]
Changed to `Go to` everywhere. Please take another look. Thanks!
